### PR TITLE
Fixing HexBlock rotation in plotBlockDiagram

### LINKED ISSUE #1421: `plotBlockDiagram()` doesn't respect block orientation
The [`plotBlockDiagram()`](https://github.com/terrapower/armi/blob/8041ad889a0203da4602f0be321b54ebef5995b6/armi/utils/plotting.py#L1375) function appears to always assume that the lattice within the block is "flats-up". This becomes an issue if one has manually defined a "points-up" 
pin grid. In that case you'll get a print like the following:
image

The problem lines appear to be in this chunk:
https://github.com/terrapower/armi/blob/8041ad889a0203da4602f0be321b54ebef5995b6/armi/utils/plotting.py#L1283-L1290

Instead of always rotating the hex patch by 30 degrees, it needs to instead be able to interrogate the underlying grid to know the rotation. Unfortunately I think this is a challenge right now, as the block itself doesn't know the type of grid that is in it (that info is stored on the associated block _blueprint_). Even still, many block blueprints don't actually have a grid attribute even if the block later gets an auto-generated grid. This is related to https://github.com/terrapower/armi/pull/730#discussion_r913942220

This type of problem could be solved once https://github.com/terrapower/armi/issues/1284 is implemented, I would guess.

### PR DESCRIPTION
**Pull request by john-science**
## What is the change?

Fixed a bug in `plotBlockDiagram()`.

## Why is the change being made?

The method `plotBlockDiagram()` incorrectly plots `HexBlock`s if they are rotated to the "corners up" rotation.

close #1421

---

## Checklist

- [x] This PR has only [one purpose or idea](https://terrapower.github.io/armi/developer/tooling.html#one-idea-one-pr).
- [x] [Tests](https://terrapower.github.io/armi/developer/tooling.html#test-it) have been added/updated to verify any new/changed code.

- [x] The code style follows [good practices](https://terrapower.github.io/armi/developer/standards_and_practices.html).
- [x] The commit message(s) follow [good practices](https://terrapower.github.io/armi/developer/tooling.html).

- [x] The [release notes](https://terrapower.github.io/armi/developer/tooling.html#add-release-notes) have been updated if necessary.
- [x] The [documentation](https://terrapower.github.io/armi/developer/tooling.html#document-it) is still up-to-date in the `doc` folder.
- [x] The dependencies are still up-to-date in `pyproject.toml`.

### CODE REVIEW

**Comment by keckler** [comment]
First question without even looking at the code -- Can we get a picture of the correct plots?

File: armi/reactor/tests/test_blocks.py
Comment (keckler) [inline armi/reactor/tests/test_blocks.py:1901]: Can we add another test for the case where this would return `False`? Otherwise it is not clear that the new `Block.cornersUp` method is working.

File: armi/utils/plotting.py
Comment (keckler) [inline armi/utils/plotting.py:1300]: I think this one needs to be rotated as well.

File: armi/utils/plotting.py
Comment (keckler) [inline armi/utils/plotting.py:1246]: Not related to your PR, but this line says that the method accepts `DerivedShape`s. It does not appear to me that this is true.

**Review by keckler — COMMENTED** [review]
I think also this chunk might need to be made aware of the orientation:
https://github.com/terrapower/armi/pull/1926/files#diff-0c52725b9d66a2bcefb306c31e4a9479fe8c206453c22b3e4cd268a1efef3c16R1181-R1185

File: armi/utils/plotting.py
Comment (john-science) [reply armi/utils/plotting.py]: done

File: armi/utils/plotting.py
Comment (john-science) [reply armi/utils/plotting.py]: eone

File: armi/reactor/tests/test_blocks.py
Comment (john-science) [reply armi/reactor/tests/test_blocks.py]: done

**Comment by keckler** [comment]
@john-science Thank you for fixing those few comments. Please also note that there are a couple more things that would be good before approving this PR:
1. https://github.com/terrapower/armi/pull/1926#pullrequestreview-2348839742
2. https://github.com/terrapower/armi/pull/1926#issuecomment-2394104416

File: armi/reactor/blocks.py
Comment (drewj-tp) [inline armi/reactor/blocks.py:1793]: What about passing off to `HexGrid.cornersUp`?

https://github.com/terrapower/armi/blob/9dbe15c621ecac1060bdfde9715cf87697c24208/armi/reactor/grids/hexagonal.py#L107-L115

```suggestion
        return self.spatialGrid.cornersUp
```

File: armi/reactor/blocks.py
Comment (drewj-tp) [inline armi/reactor/blocks.py:1791]: I guess this is false-like? But returning `None` from a bool-like method feels weird. 

Would it make more sense to return False or error here?

**Review by drewj-tp — COMMENTED** [review]
I'm currently poking around grid like things and some things here overlapped

File: armi/reactor/blocks.py
Comment (john-science) [reply armi/reactor/blocks.py:1791]: A Python `None` is a null value. It is just as valid to use `None` for a null value when a return type is an `int`, `numpy.ndarray` or `bool`.

Seems standard enough to return an empty value from a method.

What would you rather do?  I don't want to return an error, because this does not seem like a scenario in which I want to kill a simulation.

Comment (john-science) [reply armi/reactor/blocks.py:1796]: hmmm

**Comment by john-science** [comment]
@drewj-tp Well, in defense of your idea, I am still getting this:

blockDiagram27

It appears the bug for the hexagonal shaped composites was quite easy to fix. But the coordinates for the pins are just wrong from the start. (There's nothing I can do in the plotting routine to fix that.)

**Comment by john-science** [comment]
This PR is temporarily on hold for another PR: https://github.com/terrapower/armi/pull/1947

I believe my solution in this was over-engineered to work AROUND the problem present in the other PR. Once that other PR merges, I should be able to simplify this PR.

**Comment by john-science** [comment]
@keckler and @drewj-tp 

Here I plot `armiRunSmallest.yaml` with the default `hex_corners_up`:

hex_corners_up

and here I do the same thing, but alter the reactor to be "flats up":

hex_flats_up

So, after the recent bug fix, I think we finally have success!

Whew.

**Review by keckler — APPROVED** [review]
Approved, thank you.

Please update the description in the top comment of the PR. In particular, that `HexBlock.cornersUp()` method is no longer added.

**Pull request merged**
